### ISSUE TITLE
Coding: Creating Mapping Testing Pipeline

### ISSUE THREAD

**Issue by callahantiff** [callahantiff/Biolater#19]
**GOALS:** Create a pipeline that can be used to extend pediatric clinical-concept mappings to a new data source.

**Workflow Update:** @SteeleRobert has agreed to help with creating this pipeline.

<br>

**Background:**  
We have created a large set of mappings from clinical [diagnoses](https://github.com/callahantiff/Biolater/wiki/Conditions) (n=`29,128`), [medications ](https://github.com/callahantiff/Biolater/wiki/Medications)(n=`9,175` unique medications or `1,693` unique ingredients), and [measurements](https://github.com/callahantiff/Biolater/wiki/Laboratory-Tests) (n=`2,703` unique measurement results) to open biomedical ontologies. 

<br>

**TODO:**  
- Build a pipeline that performs multi-label classification.   
  - Code should take in a set of OMOP codes and output mappings, with a confidence score to a specific set of ontologies:  
    - Diagnosis codes  ➞ [Human Phenotype](https://bioportal.bioontology.org/ontologies/HP) and [Human Disease Ontologies]
(https://bioportal.bioontology.org/ontologies/DOID)  
    - Medications  ➞ [NCBITaxon](https://bioportal.bioontology.org/ontologies/NCBITAXON), [Protein Ontology](https://bioportal.bioontology.org/ontologies/PR), [ChEBI](https://bioportal.bioontology.org/ontologies/CHEBI), and [Vaccine Ontology](https://bioportal.bioontology.org/ontologies/VO)  
    - Measurement  ➞ [Human Phenotype](https://bioportal.bioontology.org/ontologies/HP)  
  - Can test code on UCHealth adult data set, MIMIC, and PIC

<br>

_General Guidelines:_  
  - Build scripts in an object-oriented framework  
    - Sketch architecture before building code  
    - General parsing class with subclasses by clinical type (i.e. conditions, measurements, medications)
  - Test-driven development
  - Needs to be written using `keras` and `TensorFlow` 
  - <u>Inputs</u>: a list of clinical codes, a list of ontologies 

**NEXT STEPS:**  
- [ ] Discuss this issue  
- [ ] Agree on roles and authorship  
- [ ] Discuss plan for moving forward, starting with discussions of architecture, prior to beginning coding

<br>

@SteeleRobert - are you good with this plan?

**Comment by SteeleRobert** [comment]
I was looking to this paper for information on multilabel classification. https://pdfs.semanticscholar.org/686d/f0720dc54c476dcb4d69584ef3c530d85f7f.pdf
I think where we can improve upon this is by trying to use the ontologies to improve clustering and ranking.

**Comment by callahantiff** [comment]
> I was looking to this paper for information on multilabel classification. https://pdfs.semanticscholar.org/686d/f0720dc54c476dcb4d69584ef3c530d85f7f.pdf
> I think where we can improve upon this is by trying to use the ontologies to improve clustering and ranking.

This looks really promising! I will make sure to read it prior to our meeting Friday so we can discuss it. Great find!
